h/api/models/document.py

```diff
@@ -26,26 +26,40 @@ class Document(Base, mixins.Timestamps):
     def __repr__(self):
         return '<Document %s>' % self.id
 
+    @property
+    def title(self):
+        titles = [m.value for m in self.meta if m.type == 'title']
+
+        if titles:
+            return titles[0]
+
     @classmethod
-    def find_or_create_by_uris(cls, claimant_uri, uris,
+    def find_by_uris(cls, session, uris):
+        """
+        Find documents by a list of uris.
+        """
+        query_uris = [text_type(uri.normalize(u), 'utf-8') for u in uris]
+
+        matching_claims = (session.query(DocumentURI).filter(
+                DocumentURI.uri_normalized.in_(query_uris))
+                .distinct(DocumentURI.document_id).subquery())
+        matching_claims_aliased = aliased(DocumentURI, matching_claims)
```

Comment:
As discussed, we don't need to make an aliased object here because we're not expecting items from the subquery in the resultset (i.e. we're not calling `session.query(Document, matching_claims_aliased)`).